WalletWasabi/Services/WalletService.cs

```diff
@@ -975,104 +942,109 @@ public Operation(Script script, Money amount, string label)
+					KeyManager.AssertLockedInternalKeysIndexed(14);
+					changeHdPubKey = KeyManager.GetKeys(KeyState.Clean, true).RandomElement();
+					builder.SetChange(changeHdPubKey.P2wpkhScript);
 				}
-
-				if (subtractFeeFromAmountIndex == i)
+				else if (payments.ChangeStrategy == ChangeStrategy.Custom)
 				{
-					realToSend[i].amount -= fee;
+					builder.SetChange(custChange.Destination);
 				}
-
-				if (realToSend[i].amount < Money.Zero)
+				else
 				{
-					throw new InsufficientBalanceException(fee + Money.Satoshis(1), realToSend[i].amount + fee);
+					throw new NotSupportedException($"{nameof(payments.ChangeStrategy)} is not supported: {payments.ChangeStrategy}.");
 				}
+				builder.SendEstimatedFees(feePerBytes);
 			}
+			var psbt = builder.BuildPSBT(false);
 
-			var toRemoveList = new List<(Script script, Money money, string label)>(realToSend);
-			toRemoveList.RemoveAll(x => x.money == Money.Zero);
-			realToSend = toRemoveList.ToArray();
+			string changeLabel = null;
+			var spentCoins = psbt.Inputs.Select(txin => smartCoinsByOutpoint[txin.PrevOut]).ToArray();
 
-			// 1. Get the possible changes.
-			Script changeScriptPubKey;
-			var sb = new StringBuilder();
-			foreach (var item in realToSend)
-			{
-				var corrected = Guard.Correct(item.label);
-				sb.Append($"{corrected}, ");
-			}
-			var changeLabel = sb.ToString().TrimEnd(',', ' ');
+			var realToSend = payments.Requests.Select(t => (label: t.Label,
+												destination: t.Destination,
+												amount: psbt.Outputs.FirstOrDefault(o => o.ScriptPubKey == t.Destination.ScriptPubKey)?.Value))
+									.Where(i => i.amount != null)
+									.ToArray();
 
-			if (customChange is null)
+			if (changeHdPubKey is HdPubKey)
 			{
-				KeyManager.AssertCleanKeysIndexed(isInternal: true);
-				KeyManager.AssertLockedInternalKeysIndexed(14);
-				var changeHdPubKey = KeyManager.GetKeys(KeyState.Clean, true).RandomElement();
-
+				var sb = new StringBuilder();
+				foreach (var item in realToSend)
+				{
+					var corrected = Guard.Correct(item.label);
+					sb.Append($"{corrected}, ");
+				}
+				changeLabel = sb.ToString().TrimEnd(',', ' ');
 				changeHdPubKey.SetLabel(changeLabel, KeyManager);
-				changeScriptPubKey = changeHdPubKey.P2wpkhScript;
 			}
-			else
+
+			if (!psbt.TryGetFee(out var fee))
 			{
-				changeScriptPubKey = customChange;
+				throw new InvalidOperationException($"Impossible to get the fees of the PSBT, this should never happen.");
 			}
+			Logger.LogInfo<WalletService>($"Fee: {fee.Satoshi} Satoshi.");
+
+			var vSize = builder.EstimateSize(psbt.GetOriginalTransaction(), true);
+			Logger.LogInfo<WalletService>($"Estimated tx size: {vSize} vbytes.");
 
 			// 6. Do some checks
 			Money totalOutgoingAmountNoFee = realToSend.Select(x => x.amount).Sum();
+			if (totalOutgoingAmountNoFee == Money.Zero)
+			{
+				throw new InvalidOperationException($"The amount to send after fee is too small to be sent to your destination");
```

Comment:
How about this message:
"The amount after `subtracting the` fee is too small to be sent`.`"
or
"The amount `without the` fee is too small to be sent`.`"